src/platform/Darwin/BleConnectionDelegateImpl.mm

```diff
@@ -115,12 +115,30 @@ - (void)centralManager:(CBCentralManager *)central
         advertisementData:(NSDictionary *)advertisementData
                      RSSI:(NSNumber *)RSSI
 {
-    BOOL isConnectable = !![advertisementData objectForKey:CBAdvertisementDataIsConnectable];
-    NSString * localNameKey = [advertisementData objectForKey:CBAdvertisementDataLocalNameKey];
-    if (isConnectable && [localNameKey isEqual:_deviceName]) {
-        ChipLogProgress(Ble, "Connecting to device: %@", _deviceName);
-        [self connect:peripheral];
-        [self stopScanning];
+    NSNumber * isConnectable = [advertisementData objectForKey:CBAdvertisementDataIsConnectable];
+    if ([isConnectable boolValue]) {
```

Comment:
supernit: how do people feel about early return and continues to decrease code indents? I find it easier to understand code if it is less indented once it starts growing, however I also see a pattern in other chip places of 'only one exit point' (but does that apply to non-resource constrained code?)

like:

```
if (![isConnectable boolValue]) {
  return;  // only connectable devices are useful
}

CBUUID * .....
for (....) {
  if (!... isEqualToData...) { 
     continue;
  }

  if (....) { 
  }
}
```

vs.

```
if ([isConnectable boolValue]) {
  CBUUID * .....;

  for (....) {
     if (...) { 
       if (...) {
       }
    }
  }
}
```

Reply:
For my part I'm 100% in favour or early returns and to declared variables just before they are used, if possible.
I think @gerickson advocates for the 'single exit point' patterns thought.